File: ProyectoElasticSearch/DemoElasticSearch/src/main/java/com/joanna/probandoelastic/controller/UserController.java
package com.joanna.probandoelastic.controller;


import com.joanna.probandoelastic.entity.User;
import com.joanna.probandoelastic.service.UserServiceImp;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/")
public class UserController {
    @Autowired
    private UserServiceImp userService;

    @PostMapping("/")
    public ResponseEntity<User> addUser(@RequestBody User user){
        User newUser = userService.addUser(user);
        return new ResponseEntity<>(newUser, HttpStatus.CREATED);
    }

    @DeleteMapping("/delete/{id}")
    public  ResponseEntity<?> delete(@PathVariable(value = "id") String id){
        User user = userService.getUserById(id);
        userService.deleteUser(user);
        return new ResponseEntity<>(HttpStatus.OK);
    }

    @PutMapping("/{id}")
    public ResponseEntity<User> updateUser(@RequestBody User user,
                                           @PathVariable(value = "id") String id){
        User newUser = userService.updateUser(user, id);
        return new ResponseEntity<>(newUser, HttpStatus.CREATED);
    }

    @GetMapping("/name/{name}")
    public ResponseEntity<User> findUserByName(@PathVariable(value = "name") String name){
        User user = userService.getUserByName(name);
        return new ResponseEntity<>(user, HttpStatus.OK);
    }


    @GetMapping("/nick/{nickname}")
    public User findUserByNickName(@PathVariable String nickname){
        return userService.getUserByNickName(nickname);
    }

    @GetMapping("/{id}")
    public User findById(@PathVariable(value="id") String id){
        return userService.getUserById(id);
    }

    @GetMapping("/all")
    public List<User> findAll(){
        return userService.getAll();
    }
}

File: kotlin/kotlinSpring/src/main/kotlin/com/joanna/kotlindemo/controller/UserController.kt
package com.joanna.kotlindemo.controller

import com.joanna.kotlindemo.entity.User
import com.joanna.kotlindemo.service.UserServiceImp
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/")
class UserController {
    @Autowired
    var userService: UserServiceImp? = null

    @PostMapping("/")
    fun addUser(@RequestBody user: User?): ResponseEntity<User?>? {
        val newUser: User = userService!!.addUser(user!!)
        return ResponseEntity<User?>(newUser, HttpStatus.CREATED)
    }

    @DeleteMapping("/delete/{name}")
    fun delete(@PathVariable(value = "name") name: String?): ResponseEntity<*>? {
        val user: User = userService!!.getUserByName(name)
        userService!!.deleteUser(user)
        return ResponseEntity<Any>(HttpStatus.OK)
    }

    @PutMapping("/{name}")
    fun updateUser(
        @RequestBody user: User?,
        @PathVariable(value = "name") name: String?
    ): ResponseEntity<User?>? {
        val newUser: User = userService!!.updateUser(user!!, name!!)
        return ResponseEntity<User?>(newUser, HttpStatus.CREATED)
    }

    @GetMapping("/name/{name}")
    fun findUserByName(@PathVariable(value = "name") name: String?): ResponseEntity<User?>? {
        val user: User = userService!!.getUserByName(name)
        return ResponseEntity<User?>(user, HttpStatus.OK)
    }


    @GetMapping("/nick/{nickname}")
    fun findUserByNickName(@PathVariable nickname: String?): User? {
        return userService!!.getUserByNickName(nickname)
    }

    @GetMapping
    fun findAll(): List<User?> {
        return userService!!.getAll()
    }
}